internal/osbuild/gcp_guest_agent_conf_stage.go

```diff
@@ -0,0 +1,104 @@
+package osbuild
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const configScopeRegex = "^(distro|instance)$"
+
+type Accounts struct {
+	DdeprovisionRemove *bool    `json:"deprovision_remove,omitempty"`
+	Groups             []string `json:"groups,omitempty"`
+	UseraddCmd         string   `json:"useradd_cmd,omitempty"`
+	UserdelCmd         string   `json:"userdel_cmd,omitempty"`
+	UsermodCmd         string   `json:"usermod_cmd,omitempty"`
+	GpasswdAddCmd      string   `json:"gpasswd_add_cmd,omitempty"`
+	GpasswdRemoveCmd   string   `json:"gpasswd_remove_cmd,omitempty"`
+	GroupaddCmd        string   `json:"groupadd_cmd,omitempty"`
+}
+
+type Daemons struct {
+	AccountsDaemon  *bool `json:"accounts_daemon,omitempty"`
+	ClockSkewDaemon *bool `json:"clock_skew_daemon,omitempty"`
+	NetworkDaemon   *bool `json:"network_daemon,omitempty"`
+}
+
+type InstanceSetup struct {
+	HostKeyTypes     []string `json:"host_key_types,omitempty"`
+	OptimizeLocalSsd *bool    `json:"optimize_local_ssd,omitempty"`
+	NetworkEnabled   *bool    `json:"network_enabled,omitempty"`
+	SetBotoConfig    *bool    `json:"set_boto_config,omitempty"`
+	SetHostKeys      *bool    `json:"set_host_keys,omitempty"`
+	SetMultiqueue    *bool    `json:"set_multiqueue,omitempty"`
+}
+
+type IpForwarding struct {
+	EthernetProtoId   string `json:"ethernet_proto_id,omitempty"`
+	IpAliases         *bool  `json:"ip_aliases,omitempty"`
+	TargetInstanceIps *bool  `json:"target_instance_ips,omitempty"`
+}
+
+type MetadataScripts struct {
+	DefaultShell string `json:"default_shell,omitempty"`
+	RunDir       string `json:"run_dir,omitempty"`
+	Startup      *bool  `json:"startup,omitempty"`
+	Shutdown     *bool  `json:"shutdown,omitempty"`
+}
+
+type NetworkInterfaces struct {
+	Setup        *bool  `json:"setup,omitempty"`
+	IpForwarding *bool  `json:"ip_forwarding,omitempty"`
+	DhcpCommand  string `json:"dhcp_command,omitempty"`
+}
+
+type Config struct {
+	Accounts          *Accounts          `json:"Accounts,omitempty"`
+	Daemons           *Daemons           `json:"Daemons,omitempty"`
+	InstanceSetup     *InstanceSetup     `json:"InstanceSetup,omitempty"`
+	IpForwarding      *IpForwarding      `json:"IpForwarding,omitempty"`
+	MetadataScripts   *MetadataScripts   `json:"MetadataScripts,omitempty"`
+	NetworkInterfaces *NetworkInterfaces `json:"NetworkInterfaces,omitempty"`
+}
+
+type GcpGuestAgentConfOptions struct {
+	ConfigScope string  `json:"config_scope,omitempty"`
```

Comment:
For enums in osbuild stage options schema, we tend to define a new type and then define the allowed values as constants to minimize the room for making a mistake (typo) when using it, although I acknowledge that the possibility of this is minimal thanks to the check in `validate()` method. An example of this is https://github.com/osbuild/osbuild-composer/blob/53d4d7f2849a8f3064f5db8468990bfe3e681e76/internal/osbuild/dnf_automatic_config_stage.go#L5-L11